### PR TITLE
Update the CSI components when their images change

### PR DESCRIPTION
**Pull request by PhanLe1010**
When users specify new images for CSI components by changing LonghornDriverDeployer environment variables, we update the corresponding CSI component to use the new images.

longhorn/longhorn#1642

### CODE REVIEW

File: csi/deployment_util.go [hunk 1]
```diff
@@ -205,6 +207,30 @@ func deploy(kubeClient *clientset.Clientset, obj runtime.Object, resource string
 	return nil
 }
 
+func needToUpdateImage(existingObj runtime.Object, newObj runtime.Object) bool {
```
Comment (c3y1huang) [inline csi/deployment_util.go:210]:
```suggestion
func needToUpdateImage(existingObj, newObj runtime.Object) bool {
```

Comment (PhanLe1010) [reply csi/deployment_util.go]: Thank you @c3y1huang . Fixed

**Review by yasker — APPROVED** [review]

**Comment by yasker** [comment]
LGTM. Let me know when the tests passed.

**Comment by PhanLe1010** [comment]
E2E core tests pass except for `test_ha_salvage ` which is a known problem.

**Pull request merged**